internal/datasource/config/config_writer.go

```diff
@@ -100,25 +99,42 @@ func (cw *ConfigWriter) Write(ctx context.Context, filesURL,
 	}
 
 	for _, fileData := range filesMetaData.GetFiles() {
-		if !doesFileRequireUpdate(cacheContent, fileData) {
+		if cacheContent != nil && !doesFileRequireUpdate(cacheContent, fileData) {
 			slog.Debug("Skipping file as latest version is already on disk", "file_path", fileData.GetPath())
 			currentFileCache[fileData.GetPath()] = cacheContent[fileData.GetPath()]
 			skippedFiles[fileData.GetPath()] = fileData
 
 			continue
 		}
 		slog.Debug("Updating file, latest version not on disk", "file_path", fileData.GetPath())
-		updateErr := cw.updateFile(ctx, fileData, filesURL, tenantID, instanceID)
-		if updateErr != nil {
+		if updateErr := cw.updateFile(ctx, fileData, filesURL, tenantID, instanceID); updateErr != nil {
 			return nil, updateErr
 		}
+		currentFileCache[fileData.GetPath()] = fileData
 	}
 
 	cw.currentFileCache = currentFileCache
 
+	if cacheContent != nil {
+		err = cw.removeFiles(currentFileCache, cacheContent)
+	}
+
 	return skippedFiles, err
 }
 
+func (cw *ConfigWriter) removeFiles(currentFileCache, fileCache CacheContent) error {
+	for _, file := range fileCache {
+		if _, ok := currentFileCache[file.GetPath()]; !ok {
+			slog.Debug("removing file", "file_path", file.GetPath())
+			if err := os.Remove(file.GetPath()); err != nil {
+				return fmt.Errorf("error removing file: %s error: %w", file.GetPath(), err)
+			}
```

Comment:
Should we add a check here for `os.IsNotExist(err))`, in case the file was already deleted before we attempt to delete it ourselves?

Reply:
Something like:
```suggestion
			if err := os.Remove(file.GetPath()); err != nil && !os.IsNotExist(err) {
				return fmt.Errorf("error removing file: %s error: %w", file.GetPath(), err)
			}
```